src/main/java/frc/robot/subsystems/Vision/VisionIO.java

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.subsystems.vision;
+
+import edu.wpi.first.math.geometry.Pose3d;
+import java.util.ArrayList;
+import java.util.List;
+import org.photonvision.targeting.PhotonTrackedTarget;
+
+/** Add your docs here. */
+public interface VisionIO {
+
+  public static class VisionIOInputs {
+    public double timestamp = 0.0;
+    public double latency = 0.0;
+    public List<PhotonTrackedTarget> targets =
+        new ArrayList<>(); // TODO make protobuf work whenever that happens
+    public double numTags = 0;
+    public Pose3d pose = new Pose3d();
```

Comment:
clarify where this is ocming from. if it can be done above the io layer that would be great. if its from the coproc clarify

Reply:
wdym where is it coming from, it should all be running on the coproc

Reply:
ye just put that in the variable name

Reply:
and maybe also that its from multi tag pnp